### PR TITLE
Improve fragment rendering logic to include JS and CSS resources

### LINKED ISSUE #129: JS, CSS resources included in a fragment are not added when the fragment is directly rendered
Consider following fragment:
```hbs
{{css "css/some-styles.css"}}
{{js "js/some-script.js"}}
<div>
     <p> some HTML content</p>
</div>
```
If this fragment rendered directly from client-side (via ajax), CSS and JS resources won't be added to the output HTML.

### PR DESCRIPTION
**Pull request by cnapagoda**
With this fix JS and CSS resources are included when a fragment is directly rendered

RESOLVE #129

### CODE REVIEW

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java [hunk 1]
```diff
@@ -68,6 +70,20 @@ private static Fragment createFragment(String name, String content) {
         return new Fragment(name, renderable, false);
     }
 
+    private static Fragment createFragmentWithResources(String name, String content) {
+        Renderable renderable = (model, l, rl, a) -> content + (model.toMap().isEmpty() ? "" : model.toMap());
+        return new Fragment(name, renderable, false) {
+            @Override
+            public String render(Model model, Lookup lookup, RequestLookup requestLookup, API api) {
+                //requestLookup.addToPlaceholder("html", content);
```
Comment (this) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: Remove this comment line.

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java [hunk 2]
```diff
@@ -68,6 +70,20 @@ private static Fragment createFragment(String name, String content) {
         return new Fragment(name, renderable, false);
     }
 
+    private static Fragment createFragmentWithResources(String name, String content) {
+        Renderable renderable = (model, l, rl, a) -> content + (model.toMap().isEmpty() ? "" : model.toMap());
```
Comment (this) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: No need to create a Renderable here as we override the render method of the Fragment.

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java [hunk 3]
```diff
@@ -68,6 +70,20 @@ private static Fragment createFragment(String name, String content) {
...
+        return new Fragment(name, renderable, false) {
```
Comment (this) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: Parameter `renderable` can replace with `null` as we override the render method of the Fragment.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java [hunk 4]
```diff
@@ -217,7 +219,16 @@ public String renderFragment(HttpRequest request, HttpResponse response) {
         Model model = new MapModel(request.getFormParams());
         RequestLookup requestLookup = createRequestLookup(request, response);
         API api = new API(sessionRegistry, requestLookup);
-        return fragment.render(model, lookup, requestLookup, api);
+
+        JsonObject renderedFragment = new JsonObject();
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: Shall we rename `renderedFragment` variable? May to `output`

Comment (cnapagoda) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: +1

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java (hunk 1 above)
Comment (cnapagoda) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: +1

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java (hunk 3 above)
Comment (cnapagoda) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: +1

File: components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java (hunk 2 above)
Comment (cnapagoda) [inline components/uuf-core/src/test/java/org/wso2/carbon/uuf/core/AppTest.java]: +1

**Pull request merged**